skimage/color/colorconv.py

```diff
@@ -789,10 +808,22 @@ def rgb2gray(rgb):
     >>> img_gray = rgb2gray(img)
     """
 
-    if rgb.ndim == 2:
+    rgb = np.asanyarray(rgb)
```

Comment:
does anybody really supply `non-arrays` as images? 

I know it is convenient to type lists of lists, but this `np.asanyarray` is quite costly to call all the time, and for such functions, I don't know if it really helps the user experience.

Reply:
It is just made to allow calling `rgb.ndim` and `rgb.shape` in the next line

Reply:
If we say in the documentation that these should be arrays (which we do I think?), then it is ok to just assume that `ndim` and `shape` are both attributes of the input.

Reply:
`image` is documented as `array_like`.